### PR TITLE
build/tools/amebasmart/gnu_utility: move pinmap_init from KM4 to AP

### PR DESCRIPTION
**Pull request by zhongnuo-tang**
1. remove pinmap_init in KM4
2. add pinmap_init in AP app start to prevent PUPD status overwritten to default due to timing issue

### CODE REVIEW

File: os/board/rtl8730e/src/component/usrcfg/rtl8730e/Make.defs [hunk 1]
```diff
@@ -52,7 +52,8 @@
 CSRCS += ameba_intfcfg.c \
 	 ameba_sleepcfg.c \
 	 ameba_wificfg.c \
-	 ameba_flashcfg.c 
+	 ameba_flashcfg.c \
+	 ameba_pinmapcfg.c
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/component/usrcfg/rtl8730e/Make.defs:56]: requires `\` at EOL

**Review by edwakuwaku — APPROVED** [review]
LGTM.

**Review by sunghan-chang — COMMENTED** [review]
@zhongnuo-tang When we merge this, os/board/rtl8730e/src/component/usrcfg/rtl8730e/ameba_pinmapcfg.c is the file to configure pin map actually used?

**Comment by zhongnuo-tang** [comment]
> @zhongnuo-tang When we merge this, os/board/rtl8730e/src/component/usrcfg/rtl8730e/ameba_pinmapcfg.c is the file to configure pin map actually used?

Hi @sunghan-chang, This is the file that configure the initial PIN status, but subsequently usage of the PIN can still modify the PUPD.

**Comment by sunghan-chang** [comment]
@zhongnuo-tang Just question (not suggestion, not request to modify),
why don't you maintain pin setting like gpio, alternative function in the same table?
If you do, we can prevent duplicated setting and can check entire pin usage.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**